R/target_market_share.R

```diff
@@ -127,102 +127,62 @@ target_market_share <- function(data,
   )
 
   check_valid_columns(data, valid_columns)
-  data <- aggregate_by_loan_id(data)
 
   crucial_scenario <- c("scenario", "tmsr", "smsp")
   check_crucial_names(scenario, crucial_scenario)
   check_crucial_names(ald, "is_ultimate_owner")
   walk_(crucial_scenario, ~ check_no_value_is_missing(scenario, .x))
 
   green_or_brown <- r2dii.data::green_or_brown
+  tmsr_or_smsp <- tmsr_or_smsp()
 
-  summary_groups <- maybe_add_name_ald(
-    c("scenario", "tmsr", "smsp", "region", "scenario_source"),
-    by_company
```

Comment:
`maybe_add_name_ald` is a function that adds (or not) the grouping variable `name_ald` to a list. In this PR, I have removed calls to this and similar functions many times, and opted to assume that `name_ald` should always be present, until the end of the script where a final summary is calculated.